flink/src/main/java/org/apache/iceberg/flink/FlinkCatalog.java

```diff
@@ -655,10 +685,54 @@ public void alterPartitionColumnStatistics(ObjectPath tablePath, CatalogPartitio
     return Lists.newArrayList(set);
   }
 
-  @Override
-  public List<CatalogPartitionSpec> listPartitions(ObjectPath tablePath, CatalogPartitionSpec partitionSpec)
-      throws CatalogException {
-    throw new UnsupportedOperationException();
+  private String getValue(Schema schema, String name, int index, StructLike structLike) {
+    Type type = schema.findType(name);
+    if (type instanceof Types.DateType) {
+      return DateTimeUtil.dateFromDays(structLike.get(index, Integer.class)).toString();
+    } else if (type instanceof Types.TimeType) {
+      return DateTimeUtil.timeFromMicros(structLike.get(index, Long.class)).toString();
+    } else if (type instanceof Types.TimestampType) {
+      return DateTimeUtil.timestampFromMicros(structLike.get(index, Long.class)).toString();
+    } else {
+      return String.valueOf(structLike.get(index, Object.class));
+    }
+  }
+
+  private org.apache.iceberg.expressions.Expression getPartitionFilter(Schema schema,
```

Comment:
Nit: It seems that the currently imported `org.apache.flink.table.expressions.Expression` is only used one time. It would possibly make sense to import `org.apache.iceberg.expressions.Expression` instead and then use the fully qualified name for the location where `org.apache.flink.table.expressions.Expression` is used.

Reply:
yes, I udpated it